steemconnect4j/src/main/java/com/hapramp/steemconnect4j/SteemConnect.java

```diff
@@ -1,7 +1,138 @@
 package com.hapramp.steemconnect4j;
 
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
 public class SteemConnect {
-	public static String getLoginUrl() {
-		return "url";
-	}
+  private SteemConnectOptions steemConnectOptions;
+
+  /**
+  * constructor for SteemConnect.
+  * @param options  options for configuration
+  */
+  public SteemConnect(SteemConnectOptions options) {
+    this.steemConnectOptions = options;
+  }
+
+  /**
+  * build login url.
+  * @return  login url for oauth.
+  */
+  public String getLoginUrl() {
+    StringBuilder loginUrlBuilder = new StringBuilder();
+    try {
+      loginUrlBuilder
+      .append(steemConnectOptions.getBaseUrl())
+          .append("/oauth2/authorize?client_id=").append(steemConnectOptions.getApp())
+          .append("&redirect_uri=")
+          .append(URLEncoder.encode(steemConnectOptions.getCallback(),"UTF-8"));
+      if (steemConnectOptions.getScope().length > 0) {
+        loginUrlBuilder
+            .append("&scope=")
+            .append(StringUtils.getCommaSeparatedString(
+            steemConnectOptions.getScope()));
+      }
+      if (steemConnectOptions.getState() != null) {
+        loginUrlBuilder.append("&state=")
+          .append(URLEncoder.encode(steemConnectOptions.getState(),"UTF-8"));
+      }
+    } catch (UnsupportedEncodingException e) {
+      e.printStackTrace();
```

Comment:
Not a good idea. If we fail to create a login URL, we should throw an exception.

```java
public String getLoginUrl() throws UnsupportedEncodingException {
    ...
}
```